src/components/templates/ScreeningRoom/ScreeningRoom.tsx

```diff
@@ -0,0 +1,153 @@
+
+  const unSelectVideo = useCallback(() => setSelectedVideo(undefined), []);
+
+  const onInputFieldChange = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      setSearchInputValue(e.target.value);
+    },
+    [setSearchInputValue]
+  );
+
+  const renderedVideoPreviews = useMemo(
+    () =>
+      videos.map((video) => (
+        <ScreeningVideoPreview
+          key={video.id}
+          video={video}
+          selectThisVideo={() => setSelectedVideo(video)}
+          selected={video.id === selectedVideoId}
+        />
+      )),
+    [videos, selectedVideoId]
+  );
+
+  const renderedCategoryOptions = useMemo(() => {
+    return (
+      <div className="ScreeningRoom__categories">
+        {categoryList.map((category) => (
+          <PosterCategory
+            key={category}
+            category={category}
+            onClick={() => setCategoryFilter(category)}
+            containerClassname="ScreeningRoom__category"
+            active={category === categoryFilter}
+          />
+        ))}
+      </div>
+    );
+  }, [categoryList, categoryFilter, setCategoryFilter]);
+
+  const renderedSubCategoryOptions = useMemo(
+    () => (
+      <div className="ScreeningRoom__subcategories">
+        {subCategoryList.map((subCategory) => (
+          <PosterCategory
+            key={subCategory}
+            category={subCategory}
+            onClick={() => setSubCategoryFilter(subCategory)}
+            containerClassname="ScreeningRoom__subcategory"
+            active={subCategory === subCategoryFilter}
+          />
+        ))}
+      </div>
+    ),
+    [subCategoryList, subCategoryFilter, setSubCategoryFilter]
+  );
+
+  return (
+    <div className="ScreeningRoom">
+      <p className="ScreeningRoom__title">Screening room</p>
+      {selectedVideo && (
+        <div className="ScreeningRoom__video-container">
+          {/* We need this additional wrapper to position the close button over the iframe */}
+          <div className="ScreeningRoom__video-cover">
+            <iframe
+              className="ScreeningRoom__video"
+              title="selected-video"
+              src={selectedVideo.videoSrc}
+              frameBorder="0"
+              allow={IFRAME_ALLOW}
+              allowFullScreen
+            />
+            <FontAwesomeIcon
+              className="ScreeningRoom__close-video-icon"
+              icon={faTimesCircle}
+              size="lg"
+              onClick={unSelectVideo}
+            />
+          </div>
+        </div>
+      )}
+      <InputField
+        containerClassName="ScreeningRoom__input-container"
+        iconStart={faSearch}
+        placeholder="Search for a talk..."
+        value={searchInputValue}
+        onChange={onInputFieldChange}
+      />
+      {renderedCategoryOptions}
+      {renderedSubCategoryOptions}
+
+      <div className="ScreeningRoom__video-previews">
+        {isVideosLoaded ? renderedVideoPreviews : "Loading videos"}
```

Comment:
Can be extracted as a const

Reply:
It's a simple ternary and we reuse this kind of codestyle across the codebase. Not sure, if this will add to readibility

Reply:
@denisdimitrov @mike-lvov FYI: For short '1 element' ternaries like this, I don't mind them being in the JSX directly since it's simple enough to 'take it all in' in a single glance. Where I have issue with using ternaries in JSX is when it's a whole giant block of code on one side or the other of it, as that becomes hard to read.

If I was to try and convert this to a non-ternary form, off the top of my head, I would probably end up moving the 'Loading videos' inside the same (presumably memo'd) code as `renderedVideoPreviews`, and return it as the result of the 'guard statement'.